crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -668,6 +669,21 @@ where
                             // threshold
                             return Some(state.finalized_block_hash)
                         }
+
+                        // OPTIMISTIC SYNCING
```

Comment:
can we feature gate this?

Reply:
or should we do this only for OP (chainspec.is_optimism) for now?

Reply:
i mean, its unlikely that it can happen on mainnet, but it can happen on testnets/devnets l1 for sure.

i can add the feature gate or check... but what would be the reason?